src/wdlci/utils/write_workflow.py

```diff
@@ -118,6 +144,15 @@ def write_workflow(
             f.write(f"\t\t{task_input}\n")
         f.write("\n")
 
+        for output_key in missing_outputs_dict.keys():
```

Comment:
I think we can check if this list is nonempty directly after making it, rather than after writing the first chunk of the file (i.e. you don't rely on anything between making the list and erroring out here to detect that there's an issue)